graph-api.yaml

```diff
@@ -0,0 +1,459 @@
+        requestBody:
+          $ref: '#/components/requestBodies/GraphQLRequest'
+        responses:
+          '200':
+            $ref: '#/components/responses/GraphQLSuccess'
+          '400':
+            $ref: '#/components/responses/GraphQLError'
+
+  /{dataset}/telicent/graphql:
+      get:
+        summary: Execute a GraphQL query against the Telicent schema
+        parameters:
+          - $ref: '#/components/parameters/dataset'
+          - $ref: '#/components/parameters/graphQLQuery'
+          - $ref: '#/components/parameters/operationName'
+          - $ref: '#/components/parameters/variables'
+          - $ref: '#/components/parameters/extensions'
+        responses:
+          '200':
+            $ref: '#/components/responses/GraphQLSuccess'
+          '400':
+            $ref: '#/components/responses/GraphQLError'
+      post:
+        summary: Execute a GraphQL query against the Telicent schema (POST)
+        parameters:
+          - $ref: '#/components/parameters/dataset'
+        requestBody:
+          $ref: '#/components/requestBodies/GraphQLRequest'
+        responses:
+          '200':
+            $ref: '#/components/responses/GraphQLSuccess'
+          '400':
+            $ref: '#/components/responses/GraphQLError'
+
+  /{dataset}/telicent/validate:
+      get:
+        summary: Validate a GraphQL query against the Telicent schema
+        parameters:
+          - $ref: '#/components/parameters/dataset'
+          - $ref: '#/components/parameters/graphQLQuery'
+          - $ref: '#/components/parameters/operationName'
+          - $ref: '#/components/parameters/variables'
+          - $ref: '#/components/parameters/extensions'
+        responses:
+          '200':
+            $ref: '#/components/responses/GraphQLSuccess'
+          '400':
+            $ref: '#/components/responses/GraphQLError'
+      post:
+        summary: Validate a GraphQL query against the Telicent schema (POST)
+        parameters:
+          - $ref: '#/components/parameters/dataset'
+        requestBody:
+          $ref: '#/components/requestBodies/GraphQLRequest'
+        responses:
+          '200':
+            $ref: '#/components/responses/GraphQLSuccess'
+          '400':
+            $ref: '#/components/responses/GraphQLError'
+
+components:
+  securitySchemes:
+    BearerAuth:
+      description: Generic Bearer Authentication is supported using JWTs
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+    AwsBearerAuth:
+      description: |
+        For AWS usage we support JWTs that are injected into requests by the ELB.  
+        Since these come in a non-standard HTTP header we have to declare this as
+        an API Key scheme in OpenAPI spec as it doesn't have a notion of using a
+        HTTP scheme in a non-standard way (yay AWS weirdness!)
+      type: apiKey
+      in: header
+      name: X-Amzn-Oidc-Data
+  parameters:
+      dataset:
+        name: dataset
+        in: query
```

Comment:
Should this be `in: path` as `dataset` is a path parameter rather than a query param?

Reply:
Good spot